lib/rbs/json_schema/generator.rb

```diff
@@ -0,0 +1,344 @@
+              end
+            end
+          else
+            raise ValidationError.new(message: "Invalid JSON Schema: #{SCHEMA_DEFINITIONS[5]}: #{schema[SCHEMA_DEFINITIONS[5]]}")
+          end
+        # SCHEMA_DEFINITIONS[2] => type => any type permitted by JSON schema
+        when schema.key?(SCHEMA_DEFINITIONS[2])
+          case schema[SCHEMA_DEFINITIONS[2]]
+          when "integer"
+            # If type = "integer", assign a class instance of integer for RBS
+            type = fetch_type("class_instance", name: "Integer", absolute: true)
+          when "number"
+            # If type = "number", assign a class instance of numeric for RBS
+            type = fetch_type("class_instance", name: "Numeric", absolute: true)
+          when "string"
+            # If type = "string", assign a class instance of string for RBS
+            type = fetch_type("class_instance", name: "String", absolute: true)
+          when "boolean"
+            # If type = "boolean", assign a base type of bool for RBS
+            type = fetch_type("boolean")
+          when "null"
+            # If type = "null", assign a base type of nil for RBS
+            type = fetch_type("nil")
+          when "object", "array"
+            # nop
+          else
+            raise ValidationError.new(message: "Invalid JSON Schema: #{SCHEMA_DEFINITIONS[2]}: #{schema[SCHEMA_DEFINITIONS[2]]}")
+          end
+        # SCHEMA_DEFINITIONS[8] => $ref => An alias type
+        when schema.key?(SCHEMA_DEFINITIONS[8])
+          string = schema[SCHEMA_DEFINITIONS[8]]
+          case
+          when string.eql?("#")
+            # If it references itself, assign alias type of itself
+            type = fetch_type("alias", name: @name)
+          when string.match?(REF_REGEX_2)
+            # If it references another file, assign alias type of reference file
+            type = fetch_type("alias", name: string.gsub(/\.json$/, ''), prefix: false)
+          when string.match?(REF_REGEX_3)
+            # If it references definitions of another file, assign alias type of definition present in reference file
+            type = fetch_type("alias", name: string.gsub(/\.json\#\/definitions\//, '__'), prefix: false)
+          when string.match?(REF_REGEX_1)
+            # If it references a definition in the current schema, assign an alias type of that definition
+            type = fetch_type("alias", name: string.gsub(/\#\/definitions\//, ''))
+          else
+            raise ValidationError.new(message: "Could not resolve URI in: #{SCHEMA_DEFINITIONS[8]} = #{schema[SCHEMA_DEFINITIONS[8]]}")
+          end
+        end
+
+        if type.nil? && !(schema_keys = schema.keys).empty? && !(schema_keys - SCHEMA_DEFINITIONS).empty?
+          RBS.logger.warn "#{key} => Unsupported attributes: #{schema_keys.join(", ")}"
+        end
+
+        type
+      end
+
+      # Returns corresponding `RBS::Types`
+      def fetch_type(type, name: nil, absolute: nil, prefix: true)
```

Comment:
It's not clear to me why we need this method. Calling the constructor methods directly would be sufficient?